ihmc-perception/src/main/java/us/ihmc/perception/heightMap/HeightMapMessageTools.java

```diff
@@ -114,11 +71,50 @@ public static void toMessage(Mat chunkDataForMessage,
       }
    }
 
-   public static void clear(ChunkMessage messageToClear)
+   /**
+    * There is no safety for this method, in the sense that we've got an array of floats coming from the message, and we've got data to put the floats into.
+    * Its expected that the data matches the size requirements of the message.
+    * By not having a safety check (i < size()) we can drastically speed up the computational time of this method.
+    * @param heightMapMessage the message that contains the data to put into the {@link HeightMapData}
+    * @return the {@link HeightMapData} that is filled with the floats from the message
+    */
+   public static HeightMapData unpackMessageToHeightMapData(HeightMapMessage heightMapMessage)
    {
-      messageToClear.setOriginX(-1.0);
-      messageToClear.setOriginX(-1.0);
-      messageToClear.getHeights().clear();
+      if (heightMapMessage == null)
+         return null;
+
+      HeightMapData heightMapData = new HeightMapData(heightMapMessage.getCellSizeInMeters(),
+                                                      heightMapMessage.getWidthInMeters(),
+                                                      heightMapMessage.getGridCenterX(),
+                                                      heightMapMessage.getGridCenterY());
+
+      Float heightsFromMessage = heightMapMessage.getHeights();
+      heightMapData.setHeights(heightsFromMessage.toArray());
```

Comment:
Just reversing the order avoid an array allocation
```suggestion
      heightsFromMessage.toArray(heightMapData.getHeights());
```